src/settingsview.cpp

```diff
@@ -58,14 +56,33 @@ bool SettingsView::confirmDirDialog(const QString& dir, QString messageText, QSt
     return (ret == QMessageBox::Ok);
 }
 
+bool SettingsView::confirmDialogDownloadDir(QString dir) {
+    auto messageText = gt("download-dir-dialog-msg");
+    messageText = messageText.replace("{{DIRECTORY}}", dir);
+    return confirmDialog(messageText, gt("download-dir-dialog-title"));
+}
+
+bool SettingsView::confirmDialogMonitorDir(QString dir, bool clearDir) {
+    QString messageText, messageTitle;
+    if (clearDir) {
+        messageText = gt("monitor-clear-dir-dialog-msg");
+        messageTitle = gt("monitor-clear-dir-dialog-title");
+    } else {
+        messageText = gt("monitor-dir-dialog-msg");
+        messageText = messageText.replace("{{DIRECTORY}}", dir);
+        messageTitle = gt("monitor-dir-dialog-title");
+    }
+    return confirmDialog(messageText, messageTitle);
+}
```

Comment:
In my previous review I missed that for the monitor-directory we have two separate messages, and one of them isn't parameterized. This helper isn't justified for the latter case where `confirmDialog()` can be called directly. Please drop the `clearDir` argument and make this wrapper similar to `confirmDialogDownloadDir()`.